### PR TITLE
(DOCSP-38852): Update the 'Configure' page for Public Preview

### PR DESCRIPTION
**Pull request by dacharyc**
## Pull Request Info

Jira ticket: https://jira.mongodb.org/browse/DOCSP-38552

- [Configure Edge Server](https://preview-mongodbdacharyc.gatsbyjs.io/atlas-app-services/DOCSP-38852/edge-server/configure/): Update the page for the Public Preview. Add `edgectl` commands and build out new sections with additional details.

Note: There are a couple of Snooty build failures related to refs that I created in the Manage Edge Servers page PR (#728). Those will be fine once both of these PRs merge to the feature branch.

TODOS to carry over to the feature branch:
- [ ] Update query details - how to set the query, and how to perform different types of common queries - once the engineering details are finalized

### Reminder Checklist

Before merging your PR, make sure to check a few things.

- [x] Did you tag pages appropriately?
  - genre
  - programming_language
  - meta.keywords
  - meta.description
- [x] Describe your PR's changes in the Release Notes section
- [x] Create a Jira ticket for related docs-realm work, if any

### Release Notes

- **Edge Server**
  - Configure Edge Server: Update the page for the Public Preview. Add `edgectl` commands and build out new sections with additional details.

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-app-services/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/edge-server/configure.txt [hunk 1]
```diff
@@ -18,89 +18,31 @@ Configure Edge Server
...
-         authentication is already enabled in the template app configuration.
-
-   .. step:: Get Edge Server Enabled
-
-      At this time, Edge Server is in Private Preview. Your MongoDB 
-      representative must enable Edge Server for your App. Give
-      your :ref:`App Services App ID <find-app-id>` to your 
-      Product or Account Representative. They will enable Edge Server and 
-      generate an authorization secret for your App.
+For more information, refer to :ref:`create-edge-server-service`.
```
Comment (dacharyc) [inline source/edge-server/configure.txt:45]: Note for reviewer: this currently shows as the ref text instead of a title in the staged PR, and the link doesn't lead anywhere. This will be resolved when it merges to the feature branch with the related PR where I created this ref. You can disregard this issue.

File: source/edge-server/configure.txt [hunk 2]
```diff
@@ -433,6 +307,44 @@ server, as well as errors that may have occurred.
          }
```
Comment (cbullinger) [inline source/edge-server/configure.txt:307]: Can't comment on it, but I don't think we need the example status response object in an example admonition

File: source/edge-server/configure.txt [hunk 3]
```diff
@@ -441,25 +353,43 @@ Upgrade the Edge Server Version
...
-~~~~~~~~~~~~~
-
-Upgrading the Edge Server version requires the host machine to have 
-`jq <https://jqlang.github.io/jq/>`__ installed. If you do not already
-have ``jq`` installed, running an Edge Server upgrade command tells you 
-to install ``jq``.
-
 .. _edge-server-upgrade-commands:
 
 Upgrade Edge Server Commands
```
Comment (cbullinger) [inline source/edge-server/configure.txt:358]: These subheadings don't really fit this page. Maybe format this similar to the "Install and Configure the Edge Server Instance" section?

> Upgrade the Edge Server Version
> -------------------------------
> 
> .. procedure 
> 
>    .. step:: Run the Upgrade Command
>    
>    < move the Upgrade Edge Server Commands section content here >
>    
>    .. step:: Deploy Changes (or whatever...) 
>    
>    < steps 2-5 continue here >

File: source/edge-server/configure.txt [hunk 4]
```diff
@@ -468,25 +398,31 @@ Upgrade Edge Server Procedure
...
-      After upgrading, from the ``edge_server`` directory, run the command 
-      to start the Edge Server:
+      After upgrading, run the Edge Server to deploy the updates.
 
       .. code-block:: shell
 
-         make up
+         edgectl start
 
    .. step:: (Optional) Reset the Edge Server
```
Comment (cbullinger) [inline source/edge-server/configure.txt:427]: is this the only place we document how to reset the server?

**Review by cbullinger — APPROVED** [review]
mostly formatting suggestions. and I know we're crunched for time, but I'm also happy to review the content again more holistically - maybe once the other PRs are merged?

File: source/edge-server/configure.txt [hunk 5]
```diff
@@ -18,103 +18,81 @@ Configure Edge Server
...
+environments, we recommend using Kubernetes to manage your Edge Server 
+containers.
 
-   For proof-of-concept testing, you can launch an EC2 instance running 
-   Ubuntu Server 18.x. Under Network settings, add a rule to allow access 
-   via HTTP/Port 80. Free tier settings are sufficient for small PoC projects.
+If you do not have Docker installed and Swarm enabled, you will be prompted
+to resolve these dependencies prior to running the Edge Server.
 
 .. _edge-server-required-network-access:
```
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/edge-server/configure.txt:97]: Well, TIL "Wire Protocol" is an official thing! 🤦‍♀️ 

I'm trying to switch most of the "wireprotocol" mentions to use "MongoDB URI" instead, as I think that's probably a more relevant detail for folks using the docs. I'll audit for my existing "wireprotocol" mentions across the PRs I've got up and try to get all these updated to either "Wire Protocol" or some other language.

[the remark this answers is not in the document]
Comment (dacharyc) [reply source/edge-server/configure.txt:97]: Re: the scannability - it isn't really 3 scenarios or a list, per se. It's just information. I think making it a list implies things that aren't really applicable here.

File: source/edge-server/configure.txt [hunk 6]
```diff
@@ -134,273 +112,169 @@ EC2, you must open ports to accept incoming connections:
...
+      - ``--platform``: You can use Docker Compose to install and manage your
+        Edge Server in a development environment, or Kubernetes to manage
+        Edge Server in a production environment. Valid values are either
+        ``compose`` or ``kubernetes``.
 
-      .. code-block:: shell
+      Follow the prompts to download and install the Edge Server code.
 
-         wget --content-disposition https://services.cloud.mongodb.com/api/client/v2.0/tiered-sync/package/latest
+   .. step:: Complete the Edge Server Configuration Details
```
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/edge-server/configure.txt:165]: Not an optional step. The Public Preview uses the `edgectl configure` command to do some setup stuff, including exchanging the registration token for an API Key that the Edge Server instance uses to authenticate with Atlas.

But I get your point - I can tweak this heading since things have changed a bit.

File: source/edge-server/configure.txt (hunk 4 above)
Comment (dacharyc) [reply source/edge-server/configure.txt:409]: Yeah. I think we have generally only needed to document this in the context of upgrading the Edge Server version. Although if there are other scenarios where people need to reset the server, it may be worth breaking this out into a separate section. 🤔

File: source/edge-server/configure.txt [hunk 7]
```diff
@@ -496,4 +432,143 @@ Upgrade Edge Server Procedure
...
+Sync All Documents
+```````````````````
+
+You can sync all documents between Edge Server and Atlas. This uses the 
+``truepredicate`` query. Because this is a simple query, you can pass it
+as a string with the ``edgectl --query`` argument.
+
+.. code-block:: shell
+
+   edgectl --query="truepredicate"
```
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/edge-server/configure.txt:516]: I believe the two are interchangeable, so I've added the default wildcard also.

File: source/edge-server/configure.txt [hunk 8]
```diff
@@ -18,113 +18,94 @@ Configure Edge Server
...
-   .. step:: Get Edge Server Enabled
+Edge Server has been verified with the following operating systems:
 
-      At this time, Edge Server is in Private Preview. Your MongoDB 
-      representative must enable Edge Server for your App. Give
-      your :ref:`App Services App ID <find-app-id>` to your 
-      Product or Account Representative. They will enable Edge Server and 
-      generate an authorization secret for your App.
+- Ubuntu Server 18+
+- macOS 11.x and newer running on M1 or Intel 2018+ MacBook Pros
```
Comment (sudssm) [inline source/edge-server/configure.txt:84]: you can add raspberry pi os 12 (https://www.raspberrypi.com/software/operating-systems/)

File: source/edge-server/configure.txt [hunk 9]
```diff
@@ -18,113 +18,94 @@ Configure Edge Server
...
+If you do not have Docker installed and Swarm enabled, you will be prompted
+to resolve these dependencies prior to running the Edge Server.
 
 .. _edge-server-required-network-access:
 
 Required Network Access
 ~~~~~~~~~~~~~~~~~~~~~~~
 
-To accept connections from sync clients or connections via the wireprotocol, 
+To accept connections from sync clients or connections using a MongoDB URI,
```
Comment (sudssm) [inline source/edge-server/configure.txt:105]: i've often seen "mongodb uri" referred to as a "connection string" elsewhere in the docs, maybe that would help make this more clear?

File: source/edge-server/configure.txt [hunk 10]
```diff
@@ -134,271 +115,162 @@ EC2, you must open ports to accept incoming connections:
...
-Other operating systems may be possible, but have not yet been verified.
-
-.. _edge-server-hardware-requirements:
-
-Hardware Requirements
-~~~~~~~~~~~~~~~~~~~~~
-
-Edge Server benchmarking is not yet complete, so minimum required hardware
-is not yet available. 
+[PLACEHOLDER: Add a link to details about configuring TLS once engineering
```
Comment (sudssm) [inline source/edge-server/configure.txt:120]: not sure if rushil's already talked to you about this, but TLDR the user is responsible for TLS termination for HTTP (including device SDK traffic), which we'd expect them to set up via a kubernetes controller.

for wireprotocol, the user will set it up via edgectl, more info coming soon

File: source/edge-server/configure.txt [hunk 11]
```diff
@@ -134,271 +115,162 @@ EC2, you must open ports to accept incoming connections:
...
+      script to manage the installation and install required dependencies.
 
-      Use ``wget`` to get the current Edge Server code as a ``.tar`` file:
+      Use this command to install the command-line tool and the Edge Server 
+      code:
 
       .. code-block:: shell
 
-         wget --content-disposition https://services.cloud.mongodb.com/api/client/v2.0/tiered-sync/package/latest
+         curl -s http://localhost:8080/edge/install.sh | bash -s -- --app-id="<INSERT-YOUR-APP-ID-HERE>" --platform=compose
```
Comment (sudssm) [inline source/edge-server/configure.txt:153]: this should be services.cloud.mongodb.com

Comment (sudssm) [reply source/edge-server/configure.txt:153]: if they are copy-pasting from here, you dont need the extra params. you can just do 
```
curl -s http://localhost:8080/edge/install.sh | bash
```

Comment (sudssm) [reply source/edge-server/configure.txt:153]: the script will then prompt the user to fill them in

File: source/edge-server/configure.txt [hunk 12]
```diff
@@ -134,271 +115,162 @@ EC2, you must open ports to accept incoming connections:
...
 
-      Unzip the tar to get the files.
+      - ``http://localhost:8080/edge/install.sh``: This assumes you are
```
Comment (sudssm) [inline source/edge-server/configure.txt:155]: this is not accurate. the url should be cloud.mongodb.com, always

File: source/edge-server/configure.txt [hunk 13]
```diff
@@ -134,271 +115,162 @@ EC2, you must open ports to accept incoming connections:
...
 
-      You now have an ``edge_server`` directory containing the server files.
+      Use ``edgectl`` to configure the Edge Server instance.
 
-   .. step:: Complete the Edge Server Configuration Details
+      .. code-block:: shell
 
-      The ``edge_server`` directory contains a :file:`config.json` file you 
-      must edit with the appropriate values to configure the server.
+         edgectl configure --app-id="<INSERT-YOUR-APP-ID-HERE>" --platform=compose
```
Comment (sudssm) [inline source/edge-server/configure.txt:175]: this is `edgectl config`

File: source/edge-server/configure.txt [hunk 14]
```diff
@@ -134,271 +115,162 @@ EC2, you must open ports to accept incoming connections:
...
+           - Boolean
+           - If true, the Edge Server instance does not run a UI on the 
+             registered ``http-port``. Defaults to ``false``.
+
+         * - ``--base-url``
+           - String
+           - The URL of the Sync server to connect to. The default is 
+             ``https://services.cloud.mongodb.com``.
+
+         * - ``--package-version``
```
Comment (sudssm) [inline source/edge-server/configure.txt:236]: curious where this came from?

File: source/edge-server/configure.txt [hunk 15]
```diff
@@ -496,4 +414,143 @@ Upgrade Edge Server Procedure
...
+
+         edgectl --update-token=true
+
+      This triggers ``edgectl`` to prompt you for the token. When you provide 
+      it, the Edge Server instance exchanges this token for a secret. It uses 
+      the secret to authenticate with Atlas when it syncs data.
+
+      .. important:: Required network access
+
+         To exchange the token for an API key, ``edgectl`` must have network
```
Comment (sudssm) [inline source/edge-server/configure.txt:464]: this would be true of update too, and potentially also config

File: source/edge-server/configure.txt [hunk 16]
```diff
@@ -496,4 +414,143 @@ Upgrade Edge Server Procedure
...
+You might only want to sync a subset of documents with an Edge Server instance.
+For example, for an instance located in a specific store, you might want to
+sync only the documents relevant to that store. An RQL query for that might
+look similar to:
+
+.. code-block:: shell
+
+   edgectl --query="store_id == 42"
+
+Specify Different Queries for Multiple Collections
```
Comment (sudssm) [inline source/edge-server/configure.txt:530]: this may be subject to change in the next couple of days. the current proposal is to either allow
```
--query-file query.json
```
or
```
--query "table:store_id==42" --query "table2:truepredicate"
```

File: source/edge-server/configure.txt (hunk 9 above)
Comment (dacharyc) [reply source/edge-server/configure.txt:105]: I was trying to be explicit here because we have the two connection types - an App connection from Device SDK, and a MongoDB connection using the MongoDB URI. "connection string" is a bit generic.

File: source/edge-server/configure.txt [hunk 17]
```diff
@@ -18,113 +18,94 @@ Configure Edge Server
...
-Server application, you may host it on a cloud server for convenience.
+After you create an Edge Server instance in Atlas, you must configure the 
+hardware that runs the Edge Server instance. For most use cases, you'll 
+provision hardware on-premises to host the Edge Server instance. While you're 
+developing your Edge Server application, you may host it on a cloud server 
+for convenience.
 
-Client devices on-prem connect to Edge Server. Your Edge Server connects 
-to App Services. Clients do not connect to App Services directly.
+Client devices on-prem connect to the Edge Server instance. Your instance
```
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/edge-server/configure.txt:27]: In this case, I'm referring to the "Edge Server Service" you can create in Atlas (as opposed to a Device Sync service). We had a discussion about terminology here on Slack: https://mongodb.slack.com/archives/C048D7YV97U/p1713445204031779

Definitely open to other ways to talk about this that might be clearer if you have thoughts.

File: source/edge-server/configure.txt (hunk 10 above)
Comment (dacharyc) [reply source/edge-server/configure.txt:120]: Nope, I haven't gotten those details from Rushil yet. Since we don't have customers doing this yet, I'll add some placeholder language around this. We can build out more details if needed when customers get to this point.

File: source/edge-server/configure.txt (hunk 16 above)
Comment (dacharyc) [reply source/edge-server/configure.txt:530]: Ok, I've added a todo that I'll carry over to the feature branch to update this once the details are finalized. Thanks!

File: source/edge-server/configure.txt (hunk 11 above)
Comment (dacharyc) [reply source/edge-server/configure.txt:153]: Alright, updating - thanks!

File: source/edge-server/configure.txt [hunk 18]
```diff
@@ -18,113 +18,95 @@ Configure Edge Server
...
+After you create an Edge Server instance in Atlas, you must configure the 
+hardware that runs the Edge Server instance. For most use cases, you'll 
+provision hardware on-premises to host the Edge Server instance. While you're 
+developing your Edge Server application, you may host it on a cloud server 
+for convenience.
 
-Client devices on-prem connect to Edge Server. Your Edge Server connects 
-to App Services. Clients do not connect to App Services directly.
+Client devices on-prem connect to the Edge Server instance. Your instance 
+connects to the Edge Server Service or Device Sync Service, depending on
```
Comment (rukumar333) [inline source/edge-server/configure.txt:28]: One question, so I think technically, you can start up device sync + edge server on a triggers app/service. Are we trying to refer to any service with Device Sync enabled as a "Device Sync Service"? And not specifically services created via the Device Sync service via MMS?

File: source/edge-server/configure.txt [hunk 19]
```diff
@@ -134,271 +116,160 @@ EC2, you must open ports to accept incoming connections:
...
-             RQL query similar to ``“store_id == 42”``.
-
-         * - ``cloudSyncServerAuthSecret``
-           - The authorization secret that your Product or Account 
-             Representative provided when you contacted them to get 
-             Edge Server enabled for your App.
-
-         * - ``httpListenPort``
-           - The port on which Edge Server listens for HTTP connections. The
+         * - ``--config-path``
```
Comment (rukumar333) [inline source/edge-server/configure.txt:189]: I believe Sudarshan talked to you about this but this file will change soon so that it can only be used to set the query for the config.

File: source/edge-server/configure.txt [hunk 20]
```diff
@@ -134,271 +116,160 @@ EC2, you must open ports to accept incoming connections:
...
+             in the service. If a corresponding email/password user does not 
+             exist, it interactively prompts for a username and password to
+             create a user. Defaults to ``false``.
+
+         * - ``--disable-ui``
+           - Boolean
+           - If true, the Edge Server instance does not run a UI on the 
+             registered ``http-port``. Defaults to ``false``.
+
+         * - ``--base-url``
```
Comment (rukumar333) [inline source/edge-server/configure.txt:230]: I think we can remove this since this isn't something that customers should be using and is meant only for internal development. I believe customers can only access `https://services.cloud.mongodb.com` and there is no other URL such as QA, dev, etc.

File: source/edge-server/configure.txt (hunk 14 above)
Comment (rukumar333) [reply source/edge-server/configure.txt:235]: Yeah I think we can drop this too for now. I need to wrap up the `edgectl upgrade` ticket and then I can provide some more information here if that works?

File: source/edge-server/configure.txt [hunk 21]
```diff
@@ -407,86 +278,140 @@ To stop the server, from the ``edge_server`` directory:
...
+   :copyable: false
 
-      {
-         "status": "ACTIVE",
-         "cloud_connected": true,
-         "num_local_clients": 1,
-         "query": {
-            "Item": "truepredicate"
-         }
+   {
```
Comment (rukumar333) [inline source/edge-server/configure.txt:296]: This has changed, I can provide an updated example in a few mins.

File: source/edge-server/configure.txt [hunk 22]
```diff
@@ -496,4 +421,143 @@ Upgrade Edge Server Procedure
...
+to specify the Edge Server instance Sync query.
+
+The ``--query`` for an Edge Server instance defaults to ``"*"``, which 
+automatically syncs all data in all of the collections in your 
+:ref:`schema <schemas>`.
+
+If you would like to filter the data that syncs to your Edge Server, you have
+two options:
+
+- Pass a simple string query. For example, ``--query="truepredicate"``.
```
Comment (rukumar333) [inline source/edge-server/configure.txt:489]: I don't think this would be valid since there isn't a table name specified for the query. But @sudssm is working on this and can probably provide more details later.

File: source/edge-server/configure.txt [hunk 23]
```diff
@@ -496,4 +421,143 @@ Upgrade Edge Server Procedure
...
+- Pass an object containing one or more collections and queries to sync a
```
Comment (rukumar333) [inline source/edge-server/configure.txt:490]: Should we indicate here that this would be via `config.json`?

File: source/edge-server/configure.txt [hunk 24]
```diff
@@ -496,4 +421,143 @@ Upgrade Edge Server Procedure
...
+
+- Pass a simple string query. For example, ``--query="truepredicate"``.
+- Pass an object containing one or more collections and queries to sync a 
+  subset of your data with Edge Server. This may resemble:
+
+  .. code-block:: json
+
+     {
+        "YOUR-COLLECTION-NAME": "YOUR-QUERY",
+        "YOUR-COLLECTION-NAME": "YOUR-QUERY"
```
Comment (rukumar333) [inline source/edge-server/configure.txt:497]: So this is a bit confusing unfortunately:

* This actually will always be the table name in the schema
* For non-Edge Server services, it is possible that the table name is different than the collection name.
* For Edge Server services, the auto generated schema when enabling the service will ensure that the table name matches the collection name.
* If a customer adds a schema on their own to any Edge Service, they can put whatever name they want.

Should we say this is the table name and then that for most cases, the table name matches the schema name but not always? What do you think?

File: source/edge-server/configure.txt (hunk 18 above)
Comment (dacharyc) [reply source/edge-server/configure.txt:28]: I have language introducing the two "types" of services on the "Manage Edge Servers in Atlas" page, which you can see here on the feature branch: [Manage Edge Servers in Atlas](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/edge-server-public-preview-docs/edge-server/manage-edge-servers/)

I didn't really get into the types of App Services that can have Device Sync. I _think_ that's outside the scope of the Edge Server docs, but if you think it's important to get into these details, I have a placeholder to populate the "[Create an App Service](https://docs-atlas-staging.mongodb.com/atlas-app-services/docsworker-xlarge/edge-server-public-preview-docs/edge-server/manage-edge-servers/#create-a-service-for-edge-server)" procedure once I can test with the UI. I could add that detail there.

IMO, it's a bit messy with all the different entrypoints, so I was trying to hand-wave away the product types a little. By the time a customer get to this page, they should have already set up the relevant service based on the "Manage" page. I just left this intro at the top in case someone skips to the "Configure" page and thinks they can start here.

File: source/edge-server/configure.txt [hunk 25]
```diff
@@ -18,113 +18,95 @@ Configure Edge Server
...
 
-The host that runs Edge Server may be an EC2 instance or other cloud 
-server, or hardware that you provision on-site.
+Required Software
+~~~~~~~~~~~~~~~~~
 
-.. tip::
+Edge Server uses Docker in Swarm mode for development. In production 
+environments, we recommend using Kubernetes to manage your Edge Server 
+containers.
```
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/edge-server/configure.txt:96]: I'll reword this to remove the "recommend" language.

File: source/edge-server/configure.txt [hunk 26]
```diff
@@ -134,271 +116,160 @@ EC2, you must open ports to accept incoming connections:
...
 
-Install and Configure the Edge Server
--------------------------------------
+Install and Configure the Edge Server Instance
+----------------------------------------------
+
+After creating an Edge Server Service or Device Sync Service, you must create 
+an Edge Server instance. Each service may have many Edge Server instances. When 
+you create an Edge Server instance, you get an installation script you can use 
+to download the Edge Server code, and a token you can use to register the
```
[the remark this answers is not in the document]
Comment (dacharyc) [reply source/edge-server/configure.txt:134]: Updated!

File: source/edge-server/configure.txt (hunk 21 above)
Comment (dacharyc) [reply source/edge-server/configure.txt:296]: Ah, good catch - thanks for mentioning this! I've got a working version locally so I can update this now. But I see there are now several new statuses in this object, with different messages - i.e.:

```
      "mongoContainerStatus": {
         "message": "running"
      },
      "mongoDBStatus": {
         "message": "ready"
      },
      "syncServerContainerStatus": {
         "message": "running"
      },
      "syncServerStatus": {
         "message": "ACTIVE"
      },
```

Just curious - is there anywhere I can see what the status message options are for each of these statuses, so I can document the options and what they mean?

File: source/edge-server/configure.txt (hunk 24 above)
Comment (dacharyc) [reply source/edge-server/configure.txt:497]: Good clarification - thank you! We generally avoid referring to "tables" as this is an implementation detail that customers may associate with relational data versus documents/collections. So I'll try updating this section to be clear that this is the schema title, which will typically match the collection name. I'd like to avoid getting into more detail than that, as I think our handling of things will take care of most of these details. We can always build this content out more in the future if people are getting confused.

File: source/edge-server/configure.txt (hunk 23 above)
Comment (dacharyc) [reply source/edge-server/configure.txt:505]: I'll update this section with more details once we've got the query implementation finalized.
